src/globemap.ts

```diff
@@ -227,32 +241,45 @@ module powerbi.extensibility.visual {
             });
             const len: number = locations.length;
             for (let i: number = 0; i < len; ++i) {
-                if (typeof (locations[i]) === "string") {
-                    const place: any = locations[i].toLowerCase();
-                    const placeKey: string = place + "/" + locationType;
-                    const location: ILocation = (!_.isEmpty(categorical.X) && !_.isEmpty(categorical.Y))
-                        ? { longitude: <number>categorical.X[0].values[i] || 0, latitude: <number>categorical.Y[0].values[i] || 0 }
-                        : undefined;
-
+                if (typeof (locations[i]) === "string" || (categorical.X && categorical.Y && categorical.X.values && categorical.Y.values)) {
                     const height: number = heights[i] / maxHeight;
                     const heat: number = heats[i] / maxHeat;
+                    let place: any;
+                    let placeKey: string;
+                    let toolTipDataLocationName: string;
+                    let location: ILocation;
+
+                    if (typeof (locations[i]) === "string") {
+                        place = locations[i].toLowerCase();
+                        placeKey = place + "/" + locationType;
+                        location = (!_.isEmpty(categorical.X) && !_.isEmpty(categorical.Y))
+                            ? { longitude: <number>categorical.X[0].values[i] || 0, latitude: <number>categorical.Y[0].values[i] || 0 }
+                            : undefined;
+                        toolTipDataLocationName = categorical.Location && categorical.Location.source.displayName;
+                    } else  {
+                        place = "Unknown";
```

Comment:
Please move ```Unknown``` to a static variable in order to be able to specify value at one particular place